SingularityService/src/main/java/com/hubspot/singularity/data/SandboxManager.java

```diff
@@ -66,6 +67,94 @@ public SlaveNotFoundException(Exception e) {
     }
   }
 
+
+  private boolean isContinuationChar(byte b) {
+    return b >= (byte)0b1000_0000 && b <= (byte)0b1011_1111;
+  }
+
+  private int numberOfFollowingBytes(byte b) {
+    if (b >= (byte)0b1100_0000 && b <= (byte)0b1101_1111) {
+      return 1;
+    } else if (b >= (byte)0b1110_0000 && b <= (byte)0b1110_1111) {
+      return 2;
+    } else if (b >= (byte)0b1111_0000 && b <= (byte)0b1111_0111) {
+      return 3;
+    } else {
+      // this could be an ASCII char or a continuation byte
+      return 0;
+    }
+  }
```

Comment:
- make `isContinuationChar()` and `numberOfFollowingBytes()` static methods, since they don't depend on any local state
- make all those binary numbers `private static final byte` constants